packages/common/src/services/logging/LogService.ts

```diff
@@ -0,0 +1,38 @@
+import { injectable, multiInject, optional } from 'inversify';
+
+import { LOG_TRANSPORTER } from '../../modules/types';
+
+import type LogTransporter from './transporters/LogTransporter';
+import { LogLevel } from './LogLevel';
+
+@injectable()
+export default class LogService {
+  constructor(@optional() @multiInject(LOG_TRANSPORTER) private readonly transporters: LogTransporter[]) {}
+
+  private log(logLevel: LogLevel, scope: string, message: string, extra?: Record<string, unknown>, error?: Error) {
+    this.transporters.forEach((transporter) => {
```

Comment:
I feel like this logic can be safely moved up a level from where it's at now in the transporters.
```suggestion
    this.transporters.filter(t => t.logLevel <= logLevel).forEach((transporter) => {
```

Reply:
I purposely placed that logic in the transporter so that the implementation is able to decide what to do with the messages.

This is useful when, for example, the transporter only collects info/debug/warn logs and publishes this alongside error/fatal logs.

For example, using Sentry, you can decide to either capture all messages as logs or to do this just before capturing errors.

Reply:
Hm, interesting use case. 2 asks then:
1) Add a comment explaining why we don't filter in LogService ("Call all transporters, regardless of log level, because some transporters use custom logic like batching or grouping logs of different levels."
2) Remove logLevel field from the LogTransporter class, because it's not indicative to all implementations if different implementations may not have a clear cut, single log level

Reply:
Good point. I did both 😄